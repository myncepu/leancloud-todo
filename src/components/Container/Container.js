import React from 'react'
import PropTypes from 'prop-types'
import {SafeAreaView} from 'react-navigation'
import {TouchableWithoutFeedback, Keyboard, View} from 'react-native'

import styles from './styles'

class Container extends React.Component {
  static propTypes = {
    children: PropTypes.any,
    backgroundColor: PropTypes.string,
  }

  render() {
    const containerStyle = [styles.container]
    if (this.props.backgroundColor) {
      containerStyle.push({ backgroundColor: this.props.backgroundColor })
    }
    return (
      <SafeAreaView style={containerStyle}>
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <View style={{ flex: 1, justifyContent: 'center' }}>
            {this.props.children}
          </View>
        </TouchableWithoutFeedback>
      </SafeAreaView>
    )
  }
}

export default Container
