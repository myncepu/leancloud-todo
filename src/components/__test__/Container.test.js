import React from 'react'
import renderer from 'react-test-renderer'
import { Container, styles } from '../Container'

it('renders successfully without children', () => {
  const rendered = renderer.create(<Container />).toJSON()
  console.log('rendered', rendered)
  expect(rendered).toBeTruthy()
})

it('exports a styles object', () => {
  expect(typeof styles).toBe('object')
})
