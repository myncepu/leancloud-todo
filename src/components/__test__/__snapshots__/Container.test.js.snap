// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children props 1`] = `
<View
  collapsable={undefined}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <View />
  </View>
</View>
`;

exports[`uses the specified backgroundColor if provided 1`] = `
<View
  collapsable={undefined}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "red",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  />
</View>
`;
